src/main/java/com/uid2/admin/audit/AdminAuditModel.java

```diff
@@ -0,0 +1,55 @@
+package com.uid2.admin.audit;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import io.vertx.core.json.JsonObject;
+
+public class AdminAuditModel implements AuditModel{
+
+    public final String tableActioned;
+    public final Actions action;
+    public final String key;
+    public final String clientIP;
+    public final String user;
+    public final String hostNode;
+    public final long timeEpochSecond;
+    public final AdditionalInfoModel additionalInfo;
+
+    public AdminAuditModel(String tableActioned, Actions action, String key, String clientIP, String user,
+                           String hostNode, long timeEpochSecond, AdditionalInfoModel additionalInfo){
+        this.tableActioned = tableActioned;
+        this.action = action;
+        this.key = key;
+        this.clientIP = clientIP;
+        this.user = user;
+        this.hostNode = hostNode;
+        this.timeEpochSecond = timeEpochSecond;
+        this.additionalInfo = additionalInfo;
+    }
+
+    @Override
+    public JsonObject writeToJson() {
+        Gson gson = new GsonBuilder().serializeNulls().create();
+        return new JsonObject(gson.toJson(this));
+    }
+
+    @Override
+    public String writeToString() {
+        return "-audit " + user + "/" + clientIP + " performed a(n) " + action.toString() + " action on " + tableActioned
+                + " with key " + this.key + " at time " + timeEpochSecond + " handled by " + hostNode + "; additional info: "
+                + this.additionalInfo;
+    }
+
+    public static class AdditionalInfoModel{
```

Comment:
Do all entity types share same additional info? Should each type define its own?

Reply:
Perhaps "AdditionalInfo" is better described as a summary field. It is expected that all types should be able to use it. The summary field is intended to be a brief English description of the DB read/write, and name/hash for validation purposes.